libpromises/var_expressions.c

```diff
@@ -95,6 +96,13 @@ static size_t VarRefHash(const VarRef *ref)
     return (h & (INT_MAX - 1));
 }
 
+unsigned int VarRefHash_untyped(const void *ref,
+                                unsigned int seed ARG_UNUSED,
+                                unsigned int max)
+{
+    return VarRefHash(ref) & (max - 1);
+}
```

Comment:
You are right actually, I just copied this code from every other hash function we have. But now that I looked deeper into it, here are some strange issues I can't help but notice:

* the `max` argument  can indeed be only power of two for this to work right; but there is no documentation, nor assert. IMHO the `max` argument should work for any number, but for now I will amend a commit that `assert()`s power of two - it should be better than a comment.
* With a quick check, it seems our hash table **always uses 8192 size of buckets**. This is bad, I'll file a redmine ticket.
* The `max` argument in already existing in all our hash functions, the purpose is for getting a random number up to the size of the hash table (in case that size was dynamic). Still I think that `max` bounding should be done separately, not in our hash function.

So I'll amend a commit asserting max, and file a bug report for refactoring `hash_map.c`.